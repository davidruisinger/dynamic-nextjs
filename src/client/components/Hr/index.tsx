import styled from 'styled-components'
import { ThemedComponent } from 'ComponentTypes'

const Hr = styled.hr<ThemedComponent>`
  border: 0;
  background-color: ${props => props.theme.color_canvas_first};
  height: 1px;
`

export default Hr
