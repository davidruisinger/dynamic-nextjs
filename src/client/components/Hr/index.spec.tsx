import 'jest-styled-components'
import React from 'react'
import testRenderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import Hr from './index'
import { templateMock } from '../../services/template/__tests__/_mocks'

const tree = testRenderer
  .create(
    <ThemeProvider theme={templateMock.variables}>
      <Hr />
    </ThemeProvider>
  )
  .toJSON()

describe('<Hr />', () => {
  it('works', () => {
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('border', '0')
    expect(tree).toHaveStyleRule('height', '1px')
    expect(tree).toHaveStyleRule(
      'background-color',
      templateMock.variables.color_canvas_first
    )
  })
})
