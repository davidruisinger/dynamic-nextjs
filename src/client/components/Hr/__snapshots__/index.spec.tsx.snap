// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hr /> works 1`] = `
.c0 {
  border: 0;
  background-color: #efefef;
  height: 1px;
}

<hr
  className="c0"
/>
`;
